daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -65,35 +82,26 @@ case object VariableInProcess extends VariableState
 
 /**
  * Class for maintaining the state of a variable
- *
- * Note: stateArg, valueArg, and defaultValueExprArg are pass by name/lazy
- * values as it is necessary to postpone their evaluation beyond when the
- * VariableInstance is created. Attempting to evaluate these arguments will
- * likely trigger an expression to be evaluated, which will query the
- * VariableMap if the expression references another variable. If this is
- * occurring within a defineVariable call, the VariableMap hasn't been fully
- * created and attempting to evaluate such an expression will result in an OOLAG
- * Circular Definition Exception
  */
 object VariableInstance {
   def apply(
-    stateArg: => VariableState,
-    valueArg: => DataValuePrimitiveNullable,
+    state: VariableState = VariableUndefined,
+    value: DataValuePrimitiveNullable = DataValue.NoValue,
     rd: VariableRuntimeData,
     defaultValueExprArg: => Maybe[CompiledExpression[AnyRef]],
```

Comment:
Do we need to pass in defaultValueExprArg anymore? I think your recent changes made it so we always get the default expression from the VariableRuntimeData when it's need?

Reply:
There are a few places that it is still needed where we only have access to VariableInstances without a VariableRuntimeData.  Things like VariableInstance.reset() and forceExpressionEvaluations()

Reply:
But VariantInstance has a reference to VRD, right? Can't those use that?

Reply:
Ah, I see what you mean now.  I can just make the defaultValueExpr a class member and not worry about passing it in.  I think that makes sense. In the VariableInstance contructor we will just set the defaultValueExpr class arg to what is in the VariableRuntimeData.

Reply:
Do we even need to do that? If something needs acces to it could it just do ``vrd.defaultValueExpr``, assume defaultValueExpr is made public if it's not already? Do we need a refrence in both the VariabeInstance and the VRD?

Reply:
You are right, we don't.  I've made the change.